src/java/org/apache/cassandra/serializers/SetSerializer.java

```diff
@@ -248,4 +253,88 @@ public ByteBuffer getSliceFromSerialized(ByteBuffer collection,
             throw new MarshalException("Not enough bytes to read a set");
         }
     }
+
+    @Override
+    public int getIndexFromSerialized(ByteBuffer collection, ByteBuffer key, AbstractType<?> comparator)
+    {
+        try
+        {
+            ByteBuffer input = collection.duplicate();
+            int n = readCollectionSize(input, ByteBufferAccessor.instance, ProtocolVersion.V3);
+            int offset = sizeOfCollectionSize(n, ProtocolVersion.V3);
+            for (int i = 0; i < n; i++)
+            {
+                ByteBuffer value = readValue(input, ByteBufferAccessor.instance, offset, ProtocolVersion.V3);
+                offset += sizeOfValue(value, ByteBufferAccessor.instance, ProtocolVersion.V3);
+                int comparison = comparator.compareForCQL(value, key);
+                if (comparison == 0)
+                    return i;
+                else if (comparison > 0)
+                    // since the set is in sorted order, we know we've gone too far and the element doesn't exist
+                    return -1;
+                // else, we're before the element so continue
```

Comment:
nit: add curly brackets. I found it a bit difficult to read when being surrounded by comments.